processor/groupbyattrsprocessor/processor.go

```diff
@@ -28,13 +28,111 @@ type groupbyattrsprocessor struct {
 
 // ProcessTraces process traces and groups traces by attribute.
 func (gap *groupbyattrsprocessor) ProcessTraces(_ context.Context, td pdata.Traces) (pdata.Traces, error) {
-	// TODO
+	rss := td.ResourceSpans()
+	for i := 0; i < rss.Len(); i++ {
+		rs := rss.At(i)
+
+		extractedResourceSpans := newSpanAttributeGroups()
+
+		ilss := rs.InstrumentationLibrarySpans()
+		for j := 0; j < ilss.Len(); j++ {
+			ils := ilss.At(j)
+			for k := 0; k < ils.Spans().Len(); k++ {
+				span := ils.Spans().At(k)
+
+				groupedAnything, groupedAttrMap, nonGroupedAttrMap := gap.splitAttrMap(span.Attributes())
+				if groupedAnything {
+					mNumGroupedSpans.M(1)
+					overwriteAttributes(nonGroupedAttrMap, span.Attributes())
+				} else {
+					mNumNonGroupedSpans.M(1)
+				}
+
+				entry := extractedResourceSpans.attributeEntry(groupedAttrMap, rs.Resource())
+				entry.instrumentationLibrarySpans(ils.InstrumentationLibrary()).Spans().Append(span)
+			}
+		}
+
+		for _, ers := range extractedResourceSpans.entries {
+			ers.resourceSpans.CopyTo(rs)
+		}
```

Comment:
CopyTo overwrites (at least should) the current value. So I don't see any point in ranging. Is it a mistake or not? Seems like this doesn't work with multiple ilss (not covered by tests). From my perspective, here should be appending of ilss to the zero resized resourceSpans.

Reply:
Yeah, a mistake! For some reason I thought CopyTo works like a ranged Upsert, but it just replaces the destination with copy of current values.